### DISCUSSION TITLE
Closed coolant flow loop

### DISCUSSION THREAD

**Discussion by dmholler** [OpenModelica/OpenModelica#12295] [Q&A] [answered]
Hi, has anyone put together a closed flow loop in OME? I'm having trouble building a closed loop from the "Parallel Cooling" example in Thermal.FluidHeatFlow.Examples. Thank you for your time.

**Comment by casella** [comment]
Using what kind of fluid, compressible or incompressible?

**Reply by dmholler** [reply]
Incompressible. The solution was to add a pressure source block.

**Reply by casella** [reply]
Yes, that is always necessary, otherwise the pressure of the system is undetermined. As in real life 😄

**Reply by dmholler** [reply]
Is there a way I can set pre-determined fluid properties? I'd like to use the Modelica.Thermal.FluidHeatFlow.Media.Water class, but when I reference this class I get the error "Expected Modelica.Thermal.FluidHeatFlow.Media.Water to be a component, but found class instead."

**Reply by casella** [reply]
I'm not sure I understand your question. The design of Modelica.Media requires the medium model to be redeclared in each component. This is not very convenient, but so far all attempts at getting language constructs to propagate the medium information automatically have failed.

**Reply by dmholler** [reply]
Okay.  I understand how to redeclare a package, but how can that be converted to the component that the compiler expects?

**Reply by casella** [reply]
Not sure what your problem is, can you post the code that is not working?

**Reply by dmholler** [reply]
I'm declaring water as my working fluid:
  package working_fluid = Modelica.Media.Water.StandardWater;

I then have heated pipe components that I want to use this medium in:
  Modelica.Thermal.FluidHeatFlow.Components.HeatedPipe Core(medium = working_fluid, m = 100, T0 = 373.15, T(start = 373.15, fixed = true), h_g = 0, T0fixed = true, tapT = 100)

I don't quite understand how to use the values from the declared class, which is a record of fluid properties.

**Reply by casella** [reply, marked as answer]
You need to write 
`Modelica.Thermal.FluidHeatFlow.Components.HeatedPipe Core(redeclare package medium = working_fluid, m = 100, ...)`
replaceable classes are not parameters.
If you do it with the GUI, you get the right syntax automatically.

**Reply by dmholler** [reply]
Thanks. When I redeclare the working_fluid package, I receive an error stating "Invalid redeclaration of component medium as class." Should I be using a partial model?

**Reply by perost** [reply]
What casella is talking about is how `Modelica.Media` works, where the components typically have a replaceable `Medium` package that it uses. But that's not the case here, `Modelica.Thermal.FluidHeatFlow` instead uses mediums derived from the record `Modelica.Thermal.FluidHeatFlow.Media.Medium`.

So you can't use `Modelica.Media.Water.StandardWater`, you have to use e.g. `Modelica.Thermal.FluidHeatFlow.Media.Water` (or define your own medium by extending `Modelica.Thermal.FluidHeatFlow.Media.Medium`).

So your first attempt was correct syntactically, you were just trying to use the wrong kind of medium:
```Modelica
parameter Modelica.Thermal.FluidHeatFlow.Media.Water working_fluid;
Modelica.Thermal.FluidHeatFlow.Components.HeatedPipe Core(medium = working_fluid, m = 100, ...);
```
or
```Modelica
Modelica.Thermal.FluidHeatFlow.Components.HeatedPipe Core(medium = Modelica.Thermal.FluidHeatFlow.Media.Water(), m = 100, ...);
```

**Reply by casella** [reply]
Sorry, I missed that you were not using Modelica.Media 😅

**Comment by dmholler** [comment]
Thank you both, I appreciate your input. I was a little confused because the GUI approach doesn't seem to insert the necessary parentheses at the end of Modelica.Thermal.FluidHeatFlow.Media.Water(). This media usage from Modelica.Thermal.FluidHeatFlow seems to work. I also didn't quite understand the manual's description, but I think I have a handle on it now. Thanks again.

**Comment by casella** [comment]
@dmholler if the GUI is not creating correct Modelica code, please [open a ticket](https://github.com/OpenModelica/OpenModelica/issues/new/choose) and explain how to reproduce the issue. We'll fix it at once.
